### PR TITLE
force send user_config on service integration update

### LINKED ISSUE #888: Can't import existing datadog service integration
## What happened?

When attempting to import any existing datadog service integration, I receive the following error:

```
terraform import module.app.module.databases.aiven_service_integration.pg_reporting_datadog_integration <my project>/<my id>

Error: cannot copy api response into terraform schema: Invalid address to set: []string{"datadog_user_config"}
```

Other service integrations can be imported just fine.

## What did you expect to happen?

Expect to be able to successfully import existing datadog service integrations

## What else do we need to know?

```
terraform --version
Terraform v1.2.8
on darwin_arm64
+ provider registry.terraform.io/aiven/aiven v3.5.1
+ provider registry.terraform.io/cyrilgdn/postgresql v1.14.0
+ provider registry.terraform.io/datadog/datadog v3.14.0
+ provider registry.terraform.io/hashicorp/aws v4.28.0
+ provider registry.terraform.io/hashicorp/external v2.1.0
+ provider registry.terraform.io/hashicorp/google v4.34.0
+ provider registry.terraform.io/hashicorp/google-beta v4.34.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/strongdm/sdm v3.2.1
```

### PR DESCRIPTION
**Pull request by byashimov**
## About this change—what it does

Resolves: #888 
Add support of datadog specific user configurable settings.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,15 +9,15 @@ nav_order: 1
...
 - Update to the official name Aiven Provider for Terraform
 - Replaced older links from help.aiven.io to docs.aiven.io
 - Change Flink version to 1.15
 - Fix empty `user_config` and empty `ip_filters`
 - Fix `tools/` consistency
-
-## [3.6.1] - 2022-09-05
-
 - Add support `is_account_owner` [field](https://api.aiven.io/doc/#operation/AccountList)
+- Fix #888, support `datadog_user_config`
```
Comment (ivan-savciuc) [inline CHANGELOG.md:20]: This change log update should be under ## [MAJOR.MINOR.PATCH] - YYYY-MM-DD not related to `3.6.1`

File: internal/service/service_integration/resource_service_integration.go [hunk 2]
```diff
@@ -229,11 +239,22 @@ func resourceServiceIntegrationUpdate(ctx context.Context, d *schema.ResourceDat
 		return diag.FromErr(err)
 	}
 
+	userConfig := resourceServiceIntegrationUserConfigFromSchemaToAPI(d)
```
Comment (ivan-savciuc) [inline internal/service/service_integration/resource_service_integration.go:242]: No need for this logic here, any `array` under user configuration options follows the rules from here `convertTerraformUserConfigValueToAPICompatibleFormatArray`, and it looks like we already have this logic - sending an empty JSON list for the options that are arrays and found to be empty.

File: internal/service/service_integration/resource_service_integration_test.go [hunk 3]
```diff
@@ -507,3 +507,67 @@ func testAccCheckAivenServiceIntegrationAttributes(n string) resource.TestCheckF
 		return nil
 	}
 }
+
+func testAccServiceIntegrationDatadogWithUserConfig(name string) string {
```
Comment (ivan-savciuc) [inline internal/service/service_integration/resource_service_integration_test.go:511]: We have the following acctest naming notation `TestAccAivenServiceIntegration_XXX` please check how other tests are done and change the name.

File: internal/service/service_integration/resource_service_integration_test.go [hunk 4]
```diff
@@ -507,3 +507,67 @@ func testAccCheckAivenServiceIntegrationAttributes(n string) resource.TestCheckF
...
+		ProviderFactories: acc.TestAccProviderFactories,
+		CheckDestroy:      testAccCheckAivenServiceIntegrationResourceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccServiceIntegrationDatadogWithUserConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "integration_type", "datadog"),
+					resource.TestCheckResourceAttr(resourceName, "project", os.Getenv("AIVEN_PROJECT_NAME")),
+					resource.TestCheckResourceAttr(resourceName, "source_service_name", fmt.Sprintf("test-acc-postgres-%s", rName)),
+					resource.TestCheckResourceAttr(resourceName, "datadog_user_config.0.datadog_tags.0.tag", "lol:bar"),
```
Comment (ivan-savciuc) [inline internal/service/service_integration/resource_service_integration_test.go:568]: Would you mind also checking `datadog_user_config.0.datadog_tags.0.comment` here if it set to the correct value?

**Review by ivan-savciuc — DISMISSED** [review]
Some minor changes requested, otherwise LGTM

**Comment by ivan-savciuc** [comment]
Also, I cannot see the new go-client dependency in the go.mod file, please update one after merging the corresponding go-client PR.

File: CHANGELOG.md (hunk 1 above)
Comment (byashimov) [reply CHANGELOG.md:20]: 3.6.1 is not released yet, don't you want it be there?

File: internal/service/service_integration/resource_service_integration_test.go (hunk 3 above)
Comment (byashimov) [reply internal/service/service_integration/resource_service_integration_test.go:511]: It's a data provider, I have followed convention:
```
testAccServiceIntegrationResource
testAccServiceIntegrationKafkaConnectResource
testAccServiceIntegrationMirrorMakerResource
testAccServiceIntegrationLogs
testAccServiceIntegrationPreexistingReadReplica
testAccServiceIntegrationShouldFailResource
testAccServiceIntegrationDatadogWithUserConfig
```

**Comment by ivan-savciuc** [comment]
running acc tests https://github.com/aiven/terraform-provider-aiven/actions/runs/3044962779

**Pull request merged**